### PR TITLE
[WIP] Adding loading states to the UI

### PR DESCRIPTION
**Pull request by jeremiak**
Refs #383

### CODE REVIEW

File: static_src/components/app_page.jsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import React from 'react';
 import AppStore from '../stores/app_store.js';
 import RouteList from './route_list.jsx';
 
+window.A = AppStore;
```
Comment (msecret) [inline static_src/components/app_page.jsx]: Remember to remove before finalizing

File: static_src/components/app_page.jsx [hunk 2]
```diff
@@ -48,12 +53,15 @@ export default class AppPage extends React.Component {
 
   render() {
     let content = <h4 className="test-none_message">No app</h4>;
```
Comment (msecret) [inline static_src/components/app_page.jsx]: So just to think through, I'd guess this is something like this:

```
let content = <h4>{ (this.state.loading) ? 'Loading...' : 'No app' }</h4>

if (!this.state.loading && this.state.app.name) { // not loading and app has data
  content = 
  ...
```

Not sure if that's valid react now that I look at it. But the point would be to always show loading state if loading and show no data if app data is missing.

File: static_src/test/unit/util/cf_api.spec.js [hunk 3]
```diff
@@ -470,6 +470,29 @@ describe('cfApi', function() {
...
+      expect(fetchAppSpy).toHaveBeenCalledOnce();
+      expect(fetchAppStatsSpy).toHaveBeenCalledOnce();
+    });
+
+    it('should call the receivedAppAll app action', function (done) {
+      const guid = 'shouldCallActionCreatorGuid';
+      const actionCreatorSpy = sandbox.spy(appActions, 'receivedAppAll');
+
+      cfApi.fetchAppAll(guid).then(() => {
+        expect(actionCreatorSpy).toHaveBeenCalled();
```
Comment (jeremiak) [inline static_src/test/unit/util/cf_api.spec.js]: @msecret this test would always fail if I was expecting the spy `toHaveBeenCalledOnce` with a message that it had been called twice, but I can't figure out why.

From looking at the way this test executes and the code in `cfApi.fetchAppAll()` can you see what makes the `appActions.receivedAppAll` get called twice?

Comment (msecret) [inline static_src/test/unit/util/cf_api.spec.js]: Hmm, I can't think of anything. When I have problems with things like this, I'll put a `debugger` statement in the action or maybe the spy and then run in in the browser with `watch-test`. Then I'll try and get a good stack trace of whos calling(it doesn't always give me a good stack track). We can check it out tomorrow.

**Pull request closed without merging**